### PR TITLE
fix: date releases by publication, and correct three wrong version claims

### PR DESCRIPTION
**Pull request by gfazioli**
Two of these were spotted on the live site; the third turned up while checking them and is the one that matters most.

## 1. Every release was dated by the commit its tag points at

The timeline read `created_at`. GitHub reports that as the date of the commit the tag points at — and `release.sh` creates the GitHub Release **before** it commits the appcast + config, so each tag lands on the *previous* release's commit. The date shown was therefore always the previous release's date:

| tag | `created_at` (was shown) | `published_at` (now shown) |
|---|---|---|
| v0.16.6 | 2026-08-05 | **2026-08-20** |
| v0.16.5 | 2026-08-05 | 2026-08-05 |
| v0.16.0 | 2026-07-30 | 2026-08-04 |

That is why 0.16.1 through 0.16.6 all read "August 5, 2026": they are tagged on commits from the same two days. Dates on older releases will shift too — to the day each was actually published.

Also fixed along the way: the hook overwrote `created_at` with a formatted string, so the `Release` type declared an ISO date and held `"August 5, 2026"`. The formatted value now has its own `displayDate` field, and `published_at` is typed `string | null` because a draft release has none.

## 2. The roadmap's version was typed by hand

It said 0.16.5 with 0.16.6 shipped. It now reads `config.app.version`, which `release.sh` already bumps as part of every release, so it cannot drift again. It had drifted before — that is what the earlier "bring the public roadmap up to 0.15.0" commit was.

## 3. The JSON-LD told search engines the app runs on macOS 15.0

`config.app.minMacOS` was `15.0`. The deployment target in the pbxproj (all eight build configs) and the appcast's `sparkle:minimumSystemVersion` both say **15.6**.

Its only reader is `StructuredData`, so this never appeared on a page: every piece of visible copy — the docs index, the FAQ, four places in Welcome — already said 15.6. That is precisely why the earlier pass across the visible copy missed it, and it is the worst class of wrong claim, because it fails *after* the download: a user on 15.0–15.5 is told the app runs, while Sparkle would never offer it and the binary would not launch.

## Also

The Wi-Fi page promised the diagnostics module "planned for v0.6" while the app is on 0.16.6. Versions are assigned when a release starts, so the number is dropped rather than corrected. And the Local services bullet now describes where a hosts-file name is shown, which changed in 0.16.6.

## Verification

- `tsc --noEmit` clean, `oxlint` clean, `next build` green (25/25 static pages).
- The roadmap interpolation **renders**, checked in the built HTML rather than assumed: `["Netfox ",["$","strong",null,{"children":"0.16.6"}]," is live` — and zero occurrences of `0.16.5` left on that page.
- The JSON-LD emits `macOS 15.6 or later` in the built output.
- `displayDate` is present in the client chunk and no `release.created_at` read remains for display.
- **Not verified on screen:** the release-notes timeline renders client-side from SWR, so the dates are not in the served HTML and I could not confirm them with a fetch. The transformation itself was checked against the real API payloads (the table above is measured, not assumed). Worth a two-second look at the deployed page.

## Added after the review passes

- **The date is pinned to UTC.** A publication date is the calendar date of an event, so it must not move with the reader: v0.15.0, published 18:03 UTC, rendered as *July 31* in New Zealand. Raised by the local review pass before push, and independently by CodeRabbit here.
- **It has a test now, and the test can fail.** The formatting moved into `formatReleaseDate` so it is testable without `swr` / `nextra/compile`. Worth stating why the test file pins a zone in `globalSetup`: the first version of the test **passed with the fix removed**, because setting `process.env.TZ` in a `beforeAll` does nothing to `Intl` — the worker has already initialised ICU, and `resolvedOptions()` kept reporting `Europe/Rome`, which agrees with UTC for these timestamps. Pinned before the workers start, deleting `timeZone: 'UTC'` now turns the suite red.
- **The Wi-Fi page no longer promises hidden-network detection as future work** — the same page says twenty lines earlier that a hidden network already appears in the list. What is missing is what it means for the channel it occupies, and the sentence now says that.
- **One finding declined:** replacing "hosts file" with generic wording in the roadmap copy. The no-infra guideline is about naming *our* stack (its examples are Sparkle, framework names); a hosts file is the reader's own file and the subject of the feature. `content/tools/devices.mdx` already names it in copy that shipped with 0.16.6, and the phrase predates this PR. Reasoning is on the thread.

Gate after all of it: `tsc` clean, `oxlint` clean, `oxfmt --check` clean, jest **2 suites / 4 tests** green, `next build` green.

### CODE REVIEW

File: components/ReleaseNotes/use-release-notes.ts [hunk 1]
```diff
@@ -76,11 +79,15 @@ export function useReleaseNotes() {
...
-            }),
+            // `published_at`, NOT `created_at`. GitHub reports `created_at` as the
+            // date of the *commit the tag points at*, and the release script creates
+            // the GitHub Release before it commits the appcast + config — so the tag
+            // lands on the previous release's commit and `created_at` is that
+            // release's date. v0.16.6 read as August 5 while it shipped on August 20.
+            displayDate: new Date(release.published_at ?? release.created_at).toLocaleDateString(
+              'en-US',
+              { year: 'numeric', month: 'long', day: 'numeric' }
+            ),
```
[the remark this answers is not in the document]
Comment (gfazioli) [reply components/ReleaseNotes/use-release-notes.ts]: Agreed and done in `59fd3fd`, ahead of this review — the local pass flagged the same thing, and the measurement backs it: v0.15.0 was published 18:03 UTC and rendered as **July 31** in `Pacific/Auckland` while GitHub shows July 30.

The boundary-case test you asked for is in `4ef1eb2`, and it was worth more than expected, because **my first version of it passed with the fix removed**. Setting `process.env.TZ` in a `beforeAll` does nothing to `Intl`: by the time a test body runs the worker has already initialised ICU, and `resolvedOptions()` kept reporting `Europe/Rome` — a zone that happens to agree with UTC for these timestamps, so the assertion held either way. It is now pinned in a `globalSetup` before any worker starts, and deleting `timeZone: 'UTC'` turns the suite red (`Expected "July 30, 2026" / Received "July 31, 2026"`). Verified by actually removing the fix, not by reading the test.

The formatting also moved out of the hook into `formatReleaseDate`, so it can be tested without `swr` and `nextra/compile` in the way.

File: content/roadmap.mdx [hunk 2]
```diff
@@ -29,7 +30,7 @@ Netfox **0.16.5** is live. The current build covers:
...
 - **Link quality** — the round trip to each device, how steady it is, how often it doesn't answer, and a chart of the day. Not the device's own signal, which only the device can measure: the path between your Mac and it.
-- **Local services** — what's listening on your own Mac, including the loopback-only ports a network check structurally cannot see. Each row names the process behind it and how far it reaches, so a forgotten dev server on localhost is something you can find; where your hosts file gives the loopback address a name, that name is shown too.
+- **Local services** — what's listening on your own Mac, including the loopback-only ports a network check structurally cannot see. Each row names the process behind it and how far it reaches, so a forgotten dev server on localhost is something you can find; where your hosts file gives an address a name, that name is shown against the listeners it actually reaches, once above the group rather than on every row.
```
[the remark this answers is not in the document]
Comment (gfazioli) [reply content/roadmap.mdx:33]: Not taking this one, and the reasoning is about what the rule protects.

The guideline forbids naming **our** infrastructure in user-facing copy — its own examples are "Sparkle", "AppKit's NSEvent monitor", framework names. A hosts file is not part of Netfox's stack: it is a file on the reader's own Mac, that they created, and it is the *subject* of the feature being described. Renaming it "local name mapping" would leave the reader unable to tell which file to go and look at, which is the entire practical value of the sentence.

Two more reasons it should stay as it is: `content/tools/devices.mdx` already names it repeatedly in copy that shipped with 0.16.6 (*"Netfox reads your Mac's own hosts file"*), so changing it here alone would make the two pages disagree; and the phrase predates this PR — the line already read "where your hosts file gives the loopback address a name", and this change only corrects *where* that name is shown.

File: content/tools/wifi.mdx [hunk 3]
```diff
@@ -79,4 +79,4 @@ Two natural limits:
 
 ## What's coming
 
-The current build is the **scan + history** version. The next major module (planned for v0.6) expands the tool into proper Wi-Fi diagnostics — channel congestion warnings, hidden network detection, neighbour-AP interference analysis. See the [Roadmap](/docs/roadmap) for the full plan.
+The current build is the **scan + history** version. The next major module expands the tool into proper Wi-Fi diagnostics — channel congestion warnings, hidden network detection, neighbour-AP interference analysis. See the [Roadmap](/docs/roadmap) for the full plan.
```
[the remark this answers is not in the document]
Comment (gfazioli) [reply content/tools/wifi.mdx]: Good catch, fixed in `daf4f75`.

The page states twenty lines earlier that a hidden network already shows up — the radio sees the beacon, and only the name is withheld — so listing its detection as future work contradicted the page itself. What is genuinely missing is what a hidden AP means for the channel it occupies (it takes up spectrum like any other, and nothing currently says so), and the sentence now says that instead of promising detection that already ships.
